CHANGES.md

```diff
@@ -1,6 +1,464 @@
+* certain dereferences of `nil` values are now caught by the compiler
+  (see https://chapel-lang.org/docs/1.19/technotes/nilChecking.html)
+* added support for using type aliases in `new` expressions
+  (see https://chapel-lang.org/docs/1.19/technotes/initTypeAlias.html)
+* added support for the `init=` method, a proposed change to copy initializers
+  (see https://chapel-lang.org/docs/1.19/technotes/initequals.html)
+* added support for ranges of codepoints and slicing of strings using them
+  (see https://chapel-lang.org/docs/1.19/builtins/String.html)
+* added `string.codepoint[s]()` to access and iterate over strings by codepoint
+  (see https://chapel-lang.org/docs/1.19/builtins/String.html#String.string.codepoint
+   and https://chapel-lang.org/docs/1.19/builtins/String.html#String.string.codepoints)
+* added `string.byte[s]()` to access and iterate over strings by byte
+  (see https://chapel-lang.org/docs/1.19/builtins/String.html#String.string.byte
+   and https://chapel-lang.org/docs/1.19/builtins/String.html#String.string.bytes)
+* added automatic fences for unordered operations at task termination
+
+Feature Improvements
+--------------------
+* added support for compile-time operations on floating point `param` values
+* added config disambiguation for `chpl` arguments via module names
+  (e.g., `-sM.debug=true` can be used to set config `debug` in module `M`)
+* made scan expressions preserve the shape/domain of the input expression
+  (e.g., in `var B = + scan A;`, B will have A's shape and domain)
+* inferred-type arrays defined using range expressions now preserve indices
+  (e.g., `var A = [i in -2..2] i;` now results an array with domain {-2..2})
+* extended `reshape()` to accept loop expressions
+  (e.g., `reshape([i in 1..4] i, {1..2, 1..2})` now works)
+* string casts to integral or floating point types may now contain underscores
+  (e.g., `"1_000_000":int` is now supported)
+* string casts to integral types can now be in base 2, 8, or 16
+  (e.g., `"0xabcd":int` is now supported)
+* array initializer expressions based on range expressions are now parallelized
+* added support for declaring homogeneous tuple types using `uint`s
+  (e.g., `param d = 3: uint; var t: d*int;` is now supported)
+* added support for `where`-clauses to `extern` procedures
+* added support for return type constraints on generic routines
+
+Deprecated and Removed Features
+-------------------------------
+* removed support for `chpl-ipe`
+* removed support for `noinit` for the time being
+* deprecated `domain.member()` in favor of `domain.contains()`
+* deprecated `string.isEmptyString()` in favor of `string.isEmpty()`
+  (see https://chapel-lang.org/docs/1.19/builtins/String.html#String.string.isEmpty)
+* deprecated support for `string.uchars()` in favor of `string.codepoints()`
+* deprecated support for `string.ulength()` in favor of `string.numCodePoints()`
```

Comment:
Add:
* deprecated support for the `codePointIndex` type in favor of `codepointIndex`
* deprecated support for `codePointToString()` in favor of `codepointToString()`